class.jetpack-feature-rollout.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/**
+ * This class will handle rolling out features to a percentage of users. Should only be used for sites connected to
+ * WordPress.com since this calls home to get the percentage.
+ *
+ * This should help us to minimize breakage by rolling out at a small percentage and increasing that over time.
+ *
+ * @since 4.4.0
+ */
+class Jetpack_Feature_Rollout {
+	/**
+	 * @var Jetpack_Feature_Rollout
+	 */
+	private static $instance = null;
+
+	/**
+	 * False if the value has not been initialized with a call to WordPress.com, or an associative array if the
+	 * we have the enabled feaures and their percentages.
+	 *
+	 * @var bool|array
+	 */
+	static $features;
+
+	/**
+	 * The transient name for storing the features
+	 *
+	 * @var string
+	 */
+	const JETPACK_FEATURES_TRANSIENT = 'jetpack_features';
+
+	static function init() {
+		if ( is_null( self::$instance ) ) {
+			self::$instance = new Jetpack_Feature_Rollout();
+		}
+
+		return self::$instance;
+	}
+
+	private function __construct() {
+		self::$features = get_transient( self::JETPACK_FEATURES_TRANSIENT );
+		self::fetch_features_if_transient_false();
+	}
+
+	/**
+	 * Will fetch features from WordPress.com if the features transient is not set.
+	 */
+	static function fetch_features_if_transient_false() {
+		if ( is_null( self::$features ) || false === self::$features ) {
+			$response = wp_remote_get( 'https://jetpack.com/get-feature-rollout/' );
+			if ( 200 === (int) wp_remote_retrieve_response_code( $response ) ) {
+				$json = json_decode( wp_remote_retrieve_body( $response ), true );
+				$features = isset( $json ) ? $json : array();
+				$transient_duration = HOUR_IN_SECONDS;
+			} else {
+				// If the request failed for some reason, then initialize to an empty array and set shorter transient
+				$features = array();
```

Comment:
I think we should preserve (to what they are currently set to or the defaults flags) the flags if the endpoint for some reason returns an error.